### PR TITLE
official profiles for Geeetech printers

### PR DESCRIPTION
**Pull request by whoseyoung**
add some series of printer parameter configurations. 15 Geeetech printers, including 3 series models.
Thunder series. A high-speed printer and the maximum printing speed can reach 300 mm/s. Mizar series. A machine that has very exquisite print quality and ultra quiet printing. Other Geeetech printers.
---Request from Geeetech offical .

### CODE REVIEW

File: resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "instantiation": "true",
+    "inherits": "fdm_filament_asa",
+    "filament_flow_ratio": [
+        "0.926"
+    ],
+    "filament_max_volumetric_speed": [
+        "12"
+    ],
+    "compatible_printers": [
+    ]
```
Comment (QingZhangBambu) [inline resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json:16]: the compatible printers need be written, or you could not used this filament in Studio

File: resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "type": "filament",
+    "filament_id": "GFB98",
```
Comment (QingZhangBambu) [inline resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json:3]: Bambu Studio has strict filament_id rule, check the filament_id of BBL, the same filament uses the same filament_id. For example, the filament_id of ABS is GFSB00

File: resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "setting_id": "GFSA04",
```
Comment (QingZhangBambu) [inline resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json:4]: Bambu Studio has strict setting_id rules. The format of setting id is to insert an S in the third position of filamnet id + GEEETECH + idx. For example, the filamnet id of ABS is GFB00, and the setting id should be GFB00_GEEETECH_00. The next ABS profile should be GFB00_GEEETECH_01. Setting ids cannot be repeated.

File: resources/profiles/Geeetech/process/0.06mm Fine @Geeetech A20 0.2 nozzle.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+    "type": "process",
+    "name": "0.06mm Fine @Geeetech A20 0.2 nozzle",
+    "inherits": "fdm_process_geeetech_0.06_nozzle_0.2",
+    "from": "system",
+    "setting_id": "GP_Geeetech A20",
```
Comment (QingZhangBambu) [inline resources/profiles/Geeetech/process/0.06mm Fine @Geeetech A20 0.2 nozzle.json:6]: the format of process setting id is GP + GEEETECH + idx. For instance,  this setting id should be GP_GEEETECH_000. next should be GP_GEEETECH_001. the setting ids should not repeat

File: resources/profiles/Geeetech/process/0.08mm Extra Fine @Geeetech A30M.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    "type": "process",
+    "name": "0.08mm Extra Fine @Geeetech A30M",
+    "inherits": "fdm_process_geeetech_0.08",
+    "from": "system",
+    "setting_id": "GP_Geeetech A30M",
+    "instantiation": "true",
+    "infill_wall_overlap": "5%",
+    "travel_speed": "100",	
+    "compatible_printers": [
+		"Geeetech A30 M 0.4 nozzle"
```
Comment (QingZhangBambu) [inline resources/profiles/Geeetech/process/0.08mm Extra Fine @Geeetech A30M.json:11]: If the parameters are exactly the same, merge the profile files. for example, add "Geeetech A30 Pro 0.4 nozzle" on compatible printers. and delete the file"0.08mm Extra Fine @Geeetech A30Pro.json".

**Comment by QingZhangBambu** [comment]
Hi, some details need to be changed.
1. check the filament id of all filaments, to ensure the format
2. check the setting id of all filaments, to ensure the format
3.  check the setting id of all process profiles, to ensure the format
4. if the params is same, merge the process profile
5. add compatible printers on filament profile, or you could not used this filament in Studio

File: resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json (hunk 1 above)
Comment (whoseyoung) [reply resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json:55]: Compatible printers added

File: resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json (hunk 2 above)
Comment (whoseyoung) [reply resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json:3]: Revised filament_id as requested

File: resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json (hunk 3 above)
Comment (whoseyoung) [reply resources/profiles/Geeetech/filament/Generic ASA @Geeetech.json]: Revised filament_id as requested

File: resources/profiles/Geeetech/process/0.06mm Fine @Geeetech A20 0.2 nozzle.json (hunk 4 above)
Comment (whoseyoung) [reply resources/profiles/Geeetech/process/0.06mm Fine @Geeetech A20 0.2 nozzle.json]: The format of process setting id has been revised as requested

File: resources/profiles/Geeetech/process/0.08mm Extra Fine @Geeetech A30M.json (hunk 5 above)
Comment (whoseyoung) [reply resources/profiles/Geeetech/process/0.08mm Extra Fine @Geeetech A30M.json]: The profile files with the same parameters have been merged

**Comment by whoseyoung** [comment]
Thank you for your prompt reply. The configuration details have been modified as requested. If there are any other issues, please inform me promptly. Thank you.

**Comment by QingZhangBambu** [comment]
@whoseyoung I will modify the filament id of ABS, and do some test internal. If the test passed, profiles will be merged.

**Comment by QingZhangBambu** [comment]
@whoseyoung Hi, To passed test  i modiy some details. If you don't mind, it PR will be abondoned and it merged it by internal way. But you still be the author.  is it OK for you? Or you'd like to change it by yourself, and merged this PR

**Comment by whoseyoung** [comment]
> @whoseyoung Hi, To passed test i modiy some details. If you don't mind, it PR will be abondoned and those profile merged by internal way. But you still be the author. is it OK for you? Or you'd like to change it by yourself, and merged this PR

Okay, I agree with your plan.

**Comment by QingZhangBambu** [comment]
> > @whoseyoung Hi, To passed test i modiy some details. If you don't mind, it PR will be abondoned and those profile merged by internal way. But you still be the author. is it OK for you? Or you'd like to change it by yourself, and merged this PR
> 
> Okay, I agree with your plan.

OK，I will merge it in internal way

**Comment by whoseyoung** [comment]
> > > @whoseyoung Hi, To passed test i modiy some details. If you don't mind, it PR will be abondoned and those profile merged by internal way. But you still be the author. is it OK for you? Or you'd like to change it by yourself, and merged this PR
> > 
> > 
> > Okay, I agree with your plan.
> 
> OK，I will merge it in internal way

May I ask if the merger of this PR has been completed, and the Geeetech model configuration cannot be found in the latest version.

**Comment by QingZhangBambu** [comment]
@whoseyoung it will be added in the official version. The latest version is hotfix version

**Comment by AndPuf** [comment]
> @whoseyoung it will be added in the official public version. The latest version is the hotfix version

Hi, i'm waiting for this update... do you know, in which version we can find it?

**Comment by QingZhangBambu** [comment]
@AndPuf it is about next week.

**Pull request closed without merging**